python/python/lance/ray/sink.py

```diff
@@ -368,6 +372,21 @@ def write(
         """Passthrough the fragments to commit phase"""
         v = []
         for block in blocks:
+            # Empty sink did not have "fragment" and "schema" in block
+            if isinstance(block, pa.Table) and block.num_columns == 0:
+                continue
+
+            from ..dependencies import _PANDAS_AVAILABLE
+            from ..dependencies import pandas as pd
+
+            # check block as pandas
+            if (
+                _PANDAS_AVAILABLE
+                and isinstance(block, pd.DataFrame)
+                and len(block.columns) == 0
+            ):
+                continue
+
```

Comment:
You could use `len` to check if the block has any data in a way that is compatible with both pyarrow and pandas:

```
if len(block) == 0:
    continue
```

Reply:
Thx, it works. And code is more readable.